File: customdialog/src/main/java/com/progressView/Builder.java
package com.progressView;

import android.content.Context;
import android.support.annotation.NonNull;
import android.view.LayoutInflater;
import android.view.View;

/**
 * Created by amit on 4/2/18.
 */

public class Builder {
    private DialogParam param;

    public Builder(@NonNull Context context) {
        param = new DialogParam();
        param.context = context;
    }

    public RequestBuilder.ViewRequest view(int layoutId) {
        param.view = LayoutInflater.from(param.context).inflate(layoutId, null);
        return new RequestBuilder.ViewRequest(param);
    }

    public RequestBuilder.ViewRequest view(@NonNull View view) {
        param.view = view;
        return new RequestBuilder.ViewRequest(param);
    }

}

File: customdialog/src/main/java/com/progressView/Builder.kt
package com.progressView

import android.content.Context
import android.view.LayoutInflater
import android.view.View

/**
 * Created by amit on 4/2/18.
 */

class Builder(context: Context) {
    private val param: DialogParam = DialogParam()

    init {
        param.context = context
    }

    fun view(layoutId: Int): RequestBuilder.ViewRequest {
        param.view = LayoutInflater.from(param.context).inflate(layoutId, null)
        return RequestBuilder.ViewRequest(param)
    }

    fun view(view: View): RequestBuilder.ViewRequest {
        param.view = view
        return RequestBuilder.ViewRequest(param)
    }

}
